### DISCUSSION TITLE
这周你做了什么？2022-04-15

### DISCUSSION THREAD

**Discussion by jiacai2050** [EmacsTalk/emacstalk.github.io#14] [Show and tell]
- 可以随便讨论本周你打算做什么与 Emacs 相关的事情，甚至可以寻求帮助或反馈。
- 请记住，什么都不做也没关系！
- Happy Hacking Emacs！

**Comment by jiacai2050** [comment]
- 最近一周在用 C 写一个 GitHub stars 管理工具，作为熟悉 C 语言的方式，主要用到了 libcurl、sqlite3、jansson，主要功能都完成了，这周末再用 Emacs dynamic module API 包一下就能在 Emacs 里用了。

```
--------------------------------------------------------------------------------
 Language             Files        Lines        Blank      Comment         Code
--------------------------------------------------------------------------------
 C                        1          596           68           20          508
 C/C++ Header             1           78            0            0           78
 SQL                      1           35            4            0           31
 Makefile                 1           25            7            1           17
 Plain Text               1            2            0            0            2
--------------------------------------------------------------------------------
 Total                    5          736           79           21          636
--------------------------------------------------------------------------------
```

**Reply by Asphaltt** [reply]
重新学习 Emacs；正在从 VSCode 切换回 Emacs

**Reply by nonamestreet** [reply]
> 重新学习 Emacs；正在从 VSCode 切换回 Emacs

我也是蛮早从VS code换回Emacs了。

**Reply by jiacai2050** [reply]
- https://github.com/jiacai2050/github-star
已经放出来了

**Comment by nocLyt** [comment]
背景：用 spacemacs 比较熟，但几乎没自己的配置。
痛点：日常工作用 lsp + ccls 浏览c++代码，但写起来感觉卡卡的。所以写 c++ 大部分时间还是用 CLion，浏览代码用 Emacs. 

这周工作有点闲，想研究了怎么让 spacemacs 上的 lsp 快一点。
1. 研究 gccemacs ( 用 `--with-native-comp` build emacs ) ，然后把两台电脑升级到了 emacs-plus@29。效果还是有的，体感上 lsp 流畅了一些，但是写起来还有点小卡顿，可能是 company 的问题。
2. 升级完 gcemacs 碰到了大家常碰到的 `libgccjit.so: error: error invoking gcc driver` 问题，在 `early-init.el` 中提前更新 `LIBRARY_PATH` 的值可以解决。但是我改回用 spacemacs develop 分支并且移除 `early-init.el` 后这个错误也消失了，不知道为啥。
3. 因为 spacemacs lsp 写c++真的感觉有点慢，根据论坛上大家列的几个大牛配置挨个试了试。说实话碰到各种问题（有点忘了为啥了...）以及已经习惯了 evil 插件，最后兜兜转转又回到了 spacemacs。
4. 关于 emacs 配置：主力电脑保持 spacemacs 的工作流，在副电脑开始自己一个一个试插件，写个自己的配置。目的是获得一个轻便的满足自己需要的 lsp + ccls 开发 c++ 的环境。总感觉 Spacemacs 太重了，打开一个 c++ 工程后上下移动光标都会卡顿。我觉得可能是移动鼠标时候绑定了什么事件？或者 UI 更新的问题？
5. 这周听了采访懒猫那期的播客，果断打开 Elisp Manual 开始看，刚看到第二章 String Type。。。下周继续来更新进度。

----
update 刚又研究了一下，发现禁用掉 `smartparens-mode` 之后在浏览 c++ 工程时，光标移动操作顺滑了不少 。
因为 profile report 显示 `sp-show--pair-function` 这个函数在我大量移动光标操作后时被调用了很多遍。

**Reply by jiacai2050** [reply]
用 eglot 就好了，比 lsp-mode 丝滑不少
- https://emacstalk.github.io/post/016/

**Comment by thomjiji** [comment]
Youtube 上看 [devaslife](https://www.youtube.com/c/devaslife/) 的视频，看他 Neovim 里用的 Solarized Dark 的主题，也在 Doom Emacs 找到相应的，最近正着迷于这个主题中。只是有一个问题就是不知道 Emacs 是不是不支持 Blur 的效果，只有透明可选？当我把 Theme 设为 doom-solarized-dark 时，set-frame-parameter 的 alpha 设成 97，带一点透明的效果感觉很不错，但是要是能再来点 Blur 就更好了。而且 doom-solarized-dark 有一个 bug：visual mode 时，code block 里选中的文字和背景完全一致，导致不能直观看到哪些文字选中了哪些没有选中，非常头疼。
